source/develop/release-management/features/sla/hosted-engine-new-deployment.html.md

```diff
@@ -0,0 +1,103 @@
+
+Today we are using the ovirt-hosted-engine-setup project to deploy hosted engine. The project contains the code to add storage, 
+disks and the vm. All of this code already exists in the ovirt-engine project. By reusing that code we can reduce bugs and unify
+the different engine installations. 
+
+## The change 
+
+The new process will use the current engine code to create all entities neeeded for successfull hosted engine deployment.   
+A local bootstrap vm with the engine appliance will be created on the host. After running engine setup on that vm we will use the 
+engine api to add a host (same one we are running on), storage domain, storage disks and finally a vm that will later become the 
+hosted engine vm in the engine (thus eliminating the need for importing it).   
+Once all the entities are created we can shut down the bootstrap vm, copy its disk to the disk we created using the engine, create 
+the hosted engine configuration files, start the agent and the broker and the hosted engine vm will start.   
+
+## How 
+
+All of this will be implemented using an Ansible script.  
+The current answer file will need to be converted to an Ansible vars files.   
+The configuration files will be created using Ansible templates and then packaged and copied to the correct location. 
+
+## Issues
+
+1. Host installation fails during network setup stage  
+   The reason for this is that vdsm setup removes the routing table.   
+   Workaround: rewrite the routing table after the host gets to the "non operational" state and then save the network configuration 
+   using vdsm-client. This causes the host to go to "up" state almost immediatly.   
+   Permanent solution: handle this in host deploy code.   
+2. Adding hosted engine disk takes a long time   
+   This might not be a problem if using a cow format and not raw. But this wasn't tested yet.   
+3. When using fqdn that resolves to a real ip the bootstrap vm tries to access the real ip and the installation fails 
+   Workaround: add a line to /etc/hosts on the host and the vm with the fqdn and the vm temporary ip.   
+   After successfull installation those lines need to be removed.   
+4. When hosted engine with iscsi domain the storage domain creation fails  
+   This seems like a bug in Ansible ovirt_storage_domain module which doesn't support this feature: 
+   https://www.ovirt.org/develop/release-management/features/storage/discard-after-delete/  
+   Until this is fixed we can't deploy with iscsi.  
+   Workaround: use direct api command until the Ansible module is fixed.   
+5. Gluster and fibre channel storage  
+   Not implemented   
+6. When creating the configuration archive agent fails to read it  
+   Ansible's archive module creates a compressed tar archive, the agent expects a non compressed one.   
+   Workaround: create the archive using the command module.  
+   Permanent solution: give the agent the ability to read compressed archive (add -z to the commands).  
```

Comment:
We can support compressed archives if and only if there is a cheap way to detect the compression. TAR is a streaming archive and we can easily read it from raw disk and still find the end. It is not that trivial when compression is in use. I would prefer if the archive stayed uncompressed - because both hosted engine and the engine would have to be updated otherwise.

Reply:
Ok, I removed the permanent solution, we can just use the command module.